### ISSUE TITLE
Polymake randomness not affected by `Oscar.randseed!`

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/Oscar.jl#3013]
Consider this example:
```
julia> Oscar.randseed!(1) ; P = rand_spherical_polytope(6,20) ; f_vector(P)
6-element Vector{ZZRingElem}:
 20
 171
 677
 1276
 1125
 375

julia> Oscar.randseed!(1) ; P = rand_spherical_polytope(6,20) ; f_vector(P)
6-element Vector{ZZRingElem}:
 20
 165
 629
 1162
 1017
 339
```
So the function `rand_spherical_polytope` produces different random output even though we called `Oscar.randseed!(1)` beforehand each time.

Looking at the code, it seems `rand_spherical_polytope` takes an optional `seed` kwarg. That could be used to "fix" this instances, but I think it would be helpful if `Oscar.randseed!` affected `rand_spherical_polytope`.

Looking at the implementation of `rand_spherical_polytope`, one way to "fix" this would be to initialize `seed` not to `nothing`, but instead to e.g. `rand(Int64)`. And then do the same for the other similar functions like `rand01_polytope` or `rand_box_polytope` etc. etc.

However, it seems better to me to address the underlying cause: Clearly if we don't specify a seed, then the polymake function that is called in the end gets a seed from *somewhere*. Before I was told that polymake doesn't really have a global RNG we could reset in `Oscar.randseed!`. But it does gets its seed value here from somewhere. Where? Can't we reset that instead?

**Comment by benlorenz** [comment]
polymake uses `/dev/urandom` (or something based pid + time as fallback if that is not available) for seeding. And the random sources are created during each function call, so there is nothing that we could reset or seed.
Using the julia rng as a default instead of nothing should work.

**Comment by fingolfin** [comment]
OK, so that's this code in Polymake:
```
void RandomSeed::renew()
{
   if (rfd==-2) rfd=open("/dev/urandom", O_RDONLY);
   if (rfd>=0) {
      if (data.fill_from_file(rfd)) return;
      rfd=-1;
   }
   static unsigned long counter=getpid();
   struct timeval tv;
   gettimeofday(&tv, nullptr);
   data=long(tv.tv_usec + (counter+=1000));
}
```
How about patching this in a Julia build of Polymake to install call into a callback function we provide. That callback then could return e.g. `rand(Int64)` or whatever.

**Comment by fingolfin** [comment]
@benlorenz ping

**Comment by benlorenz** [comment]
Yes we could probably add such a callback function and assign it from julia.

**Comment by benlorenz** [comment]
Fixed in Polymake 0.11.9:
```julia
julia> Oscar.randseed!(1) ; P = rand_spherical_polytope(6,20) ; f_vector(P)
6-element Vector{ZZRingElem}:
 20
 158
 586
 1068
 930
 310

julia> Oscar.randseed!(1) ; P = rand_spherical_polytope(6,20) ; f_vector(P)
6-element Vector{ZZRingElem}:
 20
 158
 586
 1068
 930
 310

julia> Oscar.randseed!(1) ; P = rand_spherical_polytope(6,20) ; f_vector(P)
6-element Vector{ZZRingElem}:
 20
 158
 586
 1068
 930
 310
```
